Packages/Ink/Editor/Core/InkImporter.cs

```diff
@@ -0,0 +1,64 @@
+using UnityEngine;
+using UnityEditor.AssetImporters;
+using System.IO;
+using Ink.UnityIntegration;
+using UnityEditor;
+using Ink;
+
+/// <summary>
+/// Automatically compiles .ink assets each time they are imported, and creates an InkFile asset.
+/// </summary>
+[ScriptedImporter(1, "ink")]
+public class InkImporter : ScriptedImporter
+{
+    public override void OnImportAsset(AssetImportContext ctx)
+    {
+        var inkFile = ScriptableObject.CreateInstance<InkFile>();
+        var absolutePath = InkEditorUtils.UnityRelativeToAbsolutePath(ctx.assetPath);
+        var inputString = File.ReadAllText(ctx.assetPath);
+        var compiler = new Compiler(inputString, new Compiler.Options
+        {
+            countAllVisits = true,
+            fileHandler = new UnityInkFileHandler(Path.GetDirectoryName(absolutePath)),
+            errorHandler = (string message, ErrorType type) => {
+                InkCompilerLog log;
+                if(InkCompilerLog.TryParse(message, out log)) {
+                    if(string.IsNullOrEmpty(log.relativeFilePath)) log.relativeFilePath = Path.GetFileName(absolutePath);
+                    switch (log.type)
+                    {
+                        case ErrorType.Error: 
+                            inkFile.errors.Add(log); 
+                            Debug.LogError("Ink "+log.type+" for "+Path.GetFileName(absolutePath)+": "+log.content + " (at "+log.relativeFilePath+":"+log.lineNumber+")", inkFile);
+                            break;
+                        case ErrorType.Warning: 
+                            inkFile.warnings.Add(log); 
+                            Debug.LogWarning("Ink "+log.type+" for "+Path.GetFileName(absolutePath)+": "+log.content + " (at "+log.relativeFilePath+" "+log.lineNumber+")", inkFile);
+                            break;
+                        case ErrorType.Author: 
+                            inkFile.todos.Add(log);
+                            if (InkSettings.instance.printInkLogsInConsoleOnCompile)
+                            {
+                                Debug.Log("Ink Log for "+Path.GetFileName(absolutePath)+": "+log.content + " (at "+log.relativeFilePath+" "+log.lineNumber+")", inkFile);
+                            }
+                            break;
+                    }
+                    
+                } else {
+                    Debug.LogWarning("Couldn't parse log "+message);
+                }
+            }
+        });
+        
+        try {
+            var compiledStory = compiler.Compile();
+
+            if (compiledStory != null)
+                inkFile.Initialize(compiledStory.ToJson());
+        } catch (System.SystemException e) {
+            Debug.LogException(e);
+        }
+
+        ctx.AddObjectToAsset("InkFile", inkFile);
```

Comment:
Maybe InkImporter should import something like `StoryAsset` instead? Is there a good reason for the user to handle json manually?
```c#
	public class StoryAsset : ScriptableObject
	{
		[field: SerializeField]
		public Story story { get; private set; }
		
		internal static StoryAsset Create (Story story) {
			var storyAsset = CreateInstance<StoryAsset>();
			storyAsset.story = story;
			return storyAsset;
		}
	}
```
Unless the `Story` object is designed to be modified later on by the user. I'm not that familiar with the library yet